forem/forem
# Support devcontainers and GitHub codespace

## What type of PR is this? (check all applicable)

## Description
This is a MVP implementation of devcontainers which is a development tool for vs code and also used on Github Code space!

## Related Tickets & Documents
n/a

## QA Instructions, Screenshots, Recordings
You can try this out in two ways.
1. Try it out by clicking the button of the upper right corner. This may take a minute. Once complete, you see the server running and pop-up provided to open up port 3000.
![image](https://github.com/forem/forem/assets/15793250/2ed1538a-1d4f-4e12-911d-02771e752023)

2. Try it out locally. You will need docker, vscode, and vscode's devcontainer extension. Once VS code detects `devcontainer.json` it will ask you to `rebuild this in devcontainers`. See this tutorial https://code.visualstudio.com/docs/devcontainers/tutorial
![image](https://github.com/forem/forem/assets/15793250/157a58a4-41f4-4114-bfce-da1380151d8c)

## Added tests?

## Added to documentation?

## [optional] Are there any post deployment tasks we need to perform?
n/a
